.github/workflows/main.yml

```
@@ -50,102 +127,100 @@ jobs:
-
-        - name: py313-dj52-postgres-xdist-coverage
-          python: '3.13'
-          allow_failure: false
-
-        - name: py313-dj51-postgres-xdist-coverage
-          python: '3.13'
-          allow_failure: false
-
-        - name: py312-dj42-postgres-xdist-coverage
-          python: '3.12'
-          allow_failure: false
-
-        - name: py311-dj50-postgres-xdist-coverage
-          python: '3.11'
-          allow_failure: false
-
-        - name: py311-dj42-postgres-xdist-coverage
-          python: '3.11'
-          allow_failure: false
-
-        - name: py310-dj52-postgres-xdist-coverage
-          python: '3.10'
-          allow_failure: false
-
-        - name: py310-dj51-postgres-xdist-coverage
-          python: '3.10'
-          allow_failure: false
+        run: mv .coverage coverage.${{ matrix.name }}
 
-        - name: py310-dj42-postgres-xdist-coverage
-          python: '3.10'
-          allow_failure: false
+      - name: Upload temporary coverage artifact
+        if: contains(matrix.name, 'coverage')
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-artifact-${{ matrix.name }}
+          path: coverage.${{ matrix.name }}
+          retention-days: 1
 
-        - name: py311-dj51-mysql-coverage
-          python: '3.11'
-          allow_failure: false
+  report-coverage:
+    name: Report Combined Coverage
+    runs-on: ubuntu-24.04
+    needs: test
+    if: always()
+    permissions:
+      contents: read
+      actions: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
 
-        - name: py310-dj42-mysql-coverage
-          python: '3.10'
-          allow_failure: false
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.13'
 
-        - name: py39-dj42-mysql-xdist-coverage
-          python: '3.9'
-          allow_failure: false
+      - name: Install coverage tool
+        run: python -m pip install coverage[toml]
 
-        - name: py313-djmain-sqlite-coverage
-          python: '3.13'
-          allow_failure: true
+      - name: Download all coverage artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: downloaded-coverage-artifacts
+          pattern: coverage-artifact-*
 
-        - name: py313-dj52-sqlite-coverage
-          python: '3.13'
-          allow_failure: true
+      - name: Combine coverage reports
+        run: |
+          mkdir combined_coverage_data
+          find downloaded-coverage-artifacts -type f -name 'coverage.*' -exec cp {} combined_coverage_data/ \;
+          coverage combine combined_coverage_data/*
+          coverage xml
+          coverage html
+          coverage report --format=markdown >> $GITHUB_STEP_SUMMARY
+
+      - name: Determine retention days
+        id: determine-retention-days
+        run: |
+          if [ "${{ github.ref }}" = "refs/heads/main" ] || [[ "${{ github.ref }}" == refs/tags/* ]]; then
```

To avoid [injection attacks](https://docs.zizmor.sh/audits/#template-injection):

```suggestion
          if [[ "${GITHUB_REF}" = "refs/heads/main" ]] || [[ "${GITHUB_REF}" == refs/tags/* ]]; then
```

---

Im going to make sure this doesnt happen again: https://github.com/pytest-dev/pytest-django/pull/1211

---

Woohoo! [Caught it!](https://github.com/pytest-dev/pytest-django/actions/runs/15413919234/job/43372108327?pr=1203#step:7:36)

---

Fixed!